# Chapter 4 Excercises

## Algorithm Workbench - Problems 1 and 3

Problem 1

This problem asks me to write a while loop that asks the use for a number and the number should then be multiplied by 10 and assigned to the variable product. The loop should iterate as long as the product is less than 100.

In [2]:
number = float(input('Please enter a number: ')) #This step will allow the user to input a number
product = number * 10 #I am first setting the variable product to the number inputted times 10 because the problem gave me this

while product < 100: #Here I am beginning my while loop, making sure to specify that product has to be less than 100
    print(product) #If product < 10, I want the product to be printed
    number = float(input('Please enter a number: ')) #Inputting the number and product again here will continue the loop
    product = number * 10


Please enter a number: 5
50.0
Please enter a number: 9
90.0
Please enter a number: 12


In order to create the loop, I had to state what number and product were before and after I used while, after this I can continue to enter numbers that multiplied by 10 will be less than 100 until I use a number that is too large.

Problem 3

This problem asks me to write a for loop that displays the numbers 0 - 1000 going up by 10 each time.

In [3]:
for num in range(0, 1000, 10): #I am using a for range function for this problem, I put 0 first because thats my 
                               #starting number, 1000 my ending number, and 10 as the number we're going up by
    print(num)  #For the numbers that meet all of the expectations above, the number should print below

0
10
20
30
40
50
60
70
80
90
100
110
120
130
140
150
160
170
180
190
200
210
220
230
240
250
260
270
280
290
300
310
320
330
340
350
360
370
380
390
400
410
420
430
440
450
460
470
480
490
500
510
520
530
540
550
560
570
580
590
600
610
620
630
640
650
660
670
680
690
700
710
720
730
740
750
760
770
780
790
800
810
820
830
840
850
860
870
880
890
900
910
920
930
940
950
960
970
980
990


I see that the code stopped at 990 when it was supposed to go till 1000, to fix this I will put 1001 and the last number because 1000 will then be included where when I put 1000 as the end number, it wasn't included.

In [5]:
for num in range(0, 1001, 10): #I am using a for range function for this problem, I put 0 first because thats my 
                               #starting number, 1001 is my new ending number so 1000 is included, and 10 as the number we're going up by
    print(num)  #For the numbers that meet all of the expectations above, the number should print below

0
10
20
30
40
50
60
70
80
90
100
110
120
130
140
150
160
170
180
190
200
210
220
230
240
250
260
270
280
290
300
310
320
330
340
350
360
370
380
390
400
410
420
430
440
450
460
470
480
490
500
510
520
530
540
550
560
570
580
590
600
610
620
630
640
650
660
670
680
690
700
710
720
730
740
750
760
770
780
790
800
810
820
830
840
850
860
870
880
890
900
910
920
930
940
950
960
970
980
990
1000


As you can see, when I used 1001 as the ending number, 1000 was included causing my code to run smoothly starting at 0, going up by 10, and ending at 1000.

## Programming Excercises - Problems 1, 4, 

Problem 1

This problem asks me to write a program that keeps a running total of the number of bugs collected during five days of a bug collecter is collecting them, and when the loop is finished the program should display the total number of bugs collected.

In [10]:
total = 0 #I need to set the variable total equal to 0 in order for my problem to run later and calculate the total number of bugs collected
for day in range(1, 6): #I am starting by using a for statement with the range starting at day 1 and ending at day 6
                       #So that day 5 is included but not day 6
    print('Enter the amount of bugs collected on day', day) #Here i am setting the variable bugs equal to the usering inputting how many were collected
    bugs = int(input())
    total += bugs # Setting the total += to bugs causes the program to add each amoung of bugs the user inputs as the total collected
    
print('You collected a total of', total, 'bugs') #I am printing a prompt for the program to enter the total of bugs after solving


Enter the amount of bugs collected on day 1
5
Enter the amount of bugs collected on day 2
7
Enter the amount of bugs collected on day 3
4
Enter the amount of bugs collected on day 4
2
Enter the amount of bugs collected on day 5
6
You collected a total of 24 bugs


After all these steps, the code ran asking for the number of bugs collected on days 1-5 and enterting the total collected at the end.

Problem 4

This problem asks me to write a program that asks the user for the speed of a vehicle in mph and the number of hours it has traveled. It should then use a loop to display the distance the vehicle traveled in the number of hours inputted.

In [13]:
print('This program displays a vehicles hours and distance traveled')
speed = int(input("what is the speed of the vehicle in mph? "))
time = int(input('how many hours has it traveled'))
print('hour\tdistance traveled')
print('-----------------------')
for num in range(1, time + 1):
    distance = speed * time
    print(time\tdistance)

SyntaxError: unexpected character after line continuation character (711512715.py, line 8)